### PR TITLE
Individual bonds modal functionality

### PR DESCRIPTION
**Pull request by AlexNovitchkovBurbank**
What I changed:
I added a modal with a chart in the details of each bond.

Why I changed the component:
I changed the bonds component to handle this so that I can create a candlestick chart for each chart because I feel that makes sense to it this way.

Feedback:
Does the button opening a modal make sense.

### CODE REVIEW

**Comment by shrutivd** [comment]
image

I pulled your code. Is this what the screen looks like for you as well? I can see you have integrated API in the code but I am not able to see the data on screen.  Also, I don't see your data in frames. 

It looks like you have not added any colors to the candlestick chart yet. Which is fine, we can decide in common what color scheme to go with. 

There should be a close button on the modal to close it rather than clicking somewhere else on the screen to close the modal. You can add the close button to the right side at the top.

File: src/components/bonds/individualBonds.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+function DisplayIndividualBonds(props: any) {
+  const [showModal, setShowModal] = useState(false);
+
+  const handleDetailsButtonClick = () => setShowModal(true);
+  const handleHideModal = () => setShowModal(false);
+
+  if ((props as object) === undefined) return <div></div>;
+  //if (!props.hasOwnProperty("container")) return <div></div>;
+  //if (!props.hasOwnProperty("modalChartData")) return <div></div>;
```
Comment (halltim91) [inline src/components/bonds/individualBonds.tsx:41]: If these are no longer necessary, these commented out lines need to be cleaned up

File: src/components/bonds/individualBonds.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    xaxis: {
+      autorange: true,
+      domain: [0, 1],
+      title: 'Date',
+      type: 'date',
+    },
+    yaxis: {
+      autorange: false,
+      domain: [0, 1],
+      range: [minPriceToShowOnAxis, maxPriceToShowOnAxis],
```
Comment (halltim91) [inline src/components/bonds/individualBonds.tsx:80]: Plotly should be able to infer the appropriate range for the y axis from the data. Did you try it without setting the range before you did this or is it a preference? It wasn't necessary for my candlestick graph

**Review by halltim91 — APPROVED** [review]

File: src/components/bonds/individualBonds.tsx (hunk 1 above)
Comment (AlexNovitchkovBurbank) [reply src/components/bonds/individualBonds.tsx:41]: ok

File: src/components/bonds/individualBonds.tsx (hunk 2 above)
Comment (AlexNovitchkovBurbank) [reply src/components/bonds/individualBonds.tsx:80]: I want to leave some room at the bottom and top of the chart so I will have data close but now quite at the max and min areas.

Comment (AlexNovitchkovBurbank) [reply src/components/bonds/individualBonds.tsx:80]: The graph does not look as good with autorange I think.

**Pull request merged**